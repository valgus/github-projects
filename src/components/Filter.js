import React from "react";
import License from "./License";
import PropTypes from "prop-types";

const Filter = ({ licenses = [], onChange = f => f }) => {
  return licenses.map((license, i) => {
    return (
      <License
        key={i}
        name={license.key}
        checked={license.checked}
        onChange={() => onChange(i)}
      />
    );
  });
};

Filter.propTypes = {
  licenses: PropTypes.array.isRequired,
  onChange: PropTypes.func.isRequired
};

export default Filter;
