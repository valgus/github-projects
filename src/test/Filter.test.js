import React from 'react'
import Filter from '../components/Filter'
import * as data from './test-data'
import './setup';
import renderer from 'react-test-renderer'
describe('Filter component', () => {
    const licenses = data.licenses;
    const onchange = jest.fn();
    it('renders correctly',() => { 
        const tree = renderer.create(<Filter licenses={licenses} onChange={onchange}/>).toJSON();
        expect(tree).toMatchSnapshot()
       })
})


