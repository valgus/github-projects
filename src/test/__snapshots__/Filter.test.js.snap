// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter component renders correctly 1`] = `
Array [
  <div
    className="license-box"
  >
    <label
      className="checkbox"
    >
      <input
        checked={true}
        id="mit"
        name="mit"
        onChange={[Function]}
        type="checkbox"
      />
      mit
    </label>
  </div>,
  <div
    className="license-box"
  >
    <label
      className="checkbox"
    >
      <input
        checked={false}
        id="isc"
        name="isc"
        onChange={[Function]}
        type="checkbox"
      />
      isc
    </label>
  </div>,
  <div
    className="license-box"
  >
    <label
      className="checkbox"
    >
      <input
        checked={true}
        id="apache-2.0"
        name="apache-2.0"
        onChange={[Function]}
        type="checkbox"
      />
      apache-2.0
    </label>
  </div>,
]
`;
